src/circuit/gadget/ecc/chip/add.rs

```diff
@@ -310,11 +310,11 @@ impl Config {
                     {
                         if x_p.is_zero_vartime() {
                             // 0 + Q = Q
-                            (x_q, y_q)
+                            (*x_q, *y_q)
                         } else if x_q.is_zero_vartime() {
                             // P + 0 = P
-                            (x_p, y_p)
-                        } else if (x_q == x_p) && (y_q == -y_p) {
+                            (*x_p, *y_p)
+                        } else if (x_q == x_p) && (*y_q == -y_p) {
```

Comment:
Why does only `y_q` change here?

Reply:
Could be because `-y_p` is dereferenced?

Reply:
Yes, with some wrinkles. The LHS and RHS of `==` are [place expression contexts](https://doc.rust-lang.org/reference/expressions.html#place-expressions-and-value-expressions), which means they are implicitly borrowed. But because *both* sides are implicitly borrowed, we end up with `&y_q == &-y_p`, in which the reference levels are still mismatched (although the error message says `no implementation for `&pasta_curves::Fp == pasta_curves::Fp` and doesn't mention the implicit borrowing, which I may report as a bug).